### PR TITLE
`azurerm_log_analytics_cluster`: swap to typed sdk

### PR DESCRIPTION
**Pull request by ziyeqf**
the service has creation limit on this resource, so I'm submitting these changes in one PR.

also, because of the quota limit, I'm going to update test results one by one. Once all finished I will convert it to "ready to review"

test
---
```
❯❯ tftest loganalytics TestAccLogAnalyticsCluster_basic
=== RUN   TestAccLogAnalyticsCluster_basic
=== PAUSE TestAccLogAnalyticsCluster_basic
=== CONT  TestAccLogAnalyticsCluster_basic
--- PASS: TestAccLogAnalyticsCluster_basic (1986.40s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/loganalytics	1986.448s

❯❯ tftest loganalytics TestAccLogAnalyticsCluster_resize

=== RUN   TestAccLogAnalyticsCluster_resize
=== PAUSE TestAccLogAnalyticsCluster_resize
=== CONT  TestAccLogAnalyticsCluster_resize
--- PASS: TestAccLogAnalyticsCluster_resize (2737.96s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/loganalytics	2738.017s
❯❯ tftest loganalytics TestAccLogAnalyticsCluster_requiresImport
=== RUN   TestAccLogAnalyticsCluster_requiresImport
=== PAUSE TestAccLogAnalyticsCluster_requiresImport
=== CONT  TestAccLogAnalyticsCluster_requiresImport
--- PASS: TestAccLogAnalyticsCluster_requiresImport (2567.62s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/loganalytics	2567.681s
```

### CODE REVIEW

**Comment by ziyeqf** [comment]
The service has limit on these clusters so the progress is slow..

File: internal/services/loganalytics/log_analytics_cluster_customer_managed_key_resource.go [hunk 1]
```diff
@@ -58,6 +60,12 @@ func resourceLogAnalyticsClusterCustomerManagedKey() *pluginsdk.Resource {
 				Required:     true,
 				ValidateFunc: keyVaultValidate.NestedItemIdWithOptionalVersion,
 			},
+
+			"key_rsa_size": {
```
Comment (magodo) [inline internal/services/loganalytics/log_analytics_cluster_customer_managed_key_resource.go:64]: Can we split this into its own PR?

File: internal/services/loganalytics/log_analytics_cluster_customer_managed_key_resource_test.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 	"os"
 	"testing"
 
-	"github.com/hashicorp/go-azure-sdk/resource-manager/operationalinsights/2020-08-01/clusters"
```
Comment (magodo) [inline internal/services/loganalytics/log_analytics_cluster_customer_managed_key_resource_test.go:12]: Can we move this API upgrade to its own PR?

File: internal/services/loganalytics/log_analytics_cluster_resource.go [hunk 3]
```diff
@@ -4,237 +4,333 @@
...
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/loganalytics/validate"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
-	"github.com/hashicorp/terraform-provider-azurerm/utils"
 )
 
-func resourceLogAnalyticsCluster() *pluginsdk.Resource {
```
Comment (magodo) [inline internal/services/loganalytics/log_analytics_cluster_resource.go:28]: Can we move this untyped->typed SDK migration to its own PR?

**Review by magodo — CHANGES_REQUESTED** [review]

File: internal/services/loganalytics/log_analytics_cluster_customer_managed_key_resource.go (hunk 1 above)
Comment (ziyeqf) [reply internal/services/loganalytics/log_analytics_cluster_customer_managed_key_resource.go:64]: Because the quota limit of the service split these into different PRs will cost a very long time to run these tests..

**Comment by ziyeqf** [comment]
closed by mistake

File: .gitignore [hunk 4]
```diff
@@ -53,3 +53,7 @@ tf.zip
 
 # do not upload secrets
 .env
+
+# custom for myself
+.gitignore
```
Comment (magodo) [inline .gitignore:58]: Can we avoid to check in this?

File: internal/services/loganalytics/log_analytics_cluster_resource.go [hunk 5]
```diff
@@ -4,237 +4,272 @@
...
+				if !features.FourPointOh() {
+					return 1000
+				}
+				return 500
+			}(),
+			ValidateFunc: validation.IntInSlice([]int{500, 1000, 2000, 5000}),
+		},
 
-			"tags": tags.Schema(),
+		"availability_zones_enabled": {
```
Comment (magodo) [inline internal/services/loganalytics/log_analytics_cluster_resource.go:69]: Is this introduced by a mistake?

File: internal/services/loganalytics/log_analytics_cluster_resource.go [hunk 6]
```diff
@@ -4,237 +4,272 @@
...
+				state.Identity = flattenedIdentity
 
-	if props := model.Properties; props == nil {
-		return fmt.Errorf("`azurerm_log_analytics_cluster` %s: `Properties` is nil", *id)
-	}
+				if props := model.Properties; props != nil {
+					state.ClusterId = pointer.From(props.ClusterId)
+				}
 
-	if d.HasChange("size_gb") && model.Sku != nil && model.Sku.Capacity != nil {
```
Comment (magodo) [inline internal/services/loganalytics/log_analytics_cluster_resource.go:206]: This seems is missing in the new update function?

**Review by magodo — CHANGES_REQUESTED** [review]
Thanks for this PR!
I've taken a look through and left some comments inline, but this is mostly looking good to me 👍

**Comment by ziyeqf** [comment]
The service might have some issue these days, creating new clusters always get HTTP 500, will append new test results once the service healthy

**Comment by tombuildsstuff** [comment]
@ziyeqf as this PR's been sitting as a draft since September with no update since December I'm going to temporary close this PR for the moment - but once you're able to test this please let us know and we can reopen this/take another look.

Thanks!

**Comment by ziyeqf** [comment]
Hi @tombuildsstuff, this was blocked by a issue on the service before. The issue is gone and I'm running these tests. Can we reopen this PR?

**Comment by tombuildsstuff** [comment]
@ziyeqf so is this no longer a WIP?

**Comment by ziyeqf** [comment]
> @ziyeqf so is this no longer a WIP?

Yes, it's ready to review.

**Review by katbyte — APPROVED** [review]
LGTM 🧊

**Comment by ziyeqf** [comment]
kindly ping for merging
